dwarf/expr.cc

```diff
@@ -41,9 +41,12 @@ expr::evaluate(expr_context *ctx, const std::initializer_list<taddr> &arguments)
         // Create the initial stack.  arguments are in reverse order
         // (that is, element 0 is TOS), so reverse it.
         stack.reserve(arguments.size());
-        for (const taddr *elt = arguments.end() - 1;
-             elt >= arguments.begin(); elt--)
-                stack.push_back(*elt);
+        if (arguments.size() > 0)
+        {
+          for (const taddr *elt = arguments.end() - 1;
+               elt >= arguments.begin(); elt--)
```

Comment:
Now that you've pointed out the problem with the size==0 case, I think there's another problem here: `elt` will decrease below `begin()` before the loop exits, which is also undefined.

C++14 added `rbegin()` and `rend()`, which would be perfect for this, but the rest of the code is C++11-compatible, so it seems a little silly to use a C++14ism just here.

Maybe instead this should just be

```
for (int i = arguments.size() - 1; i >= 0; i--)
        stack.push_back(arguments[i]);
```